hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/receiver/kafka/BasicMessageContentReader.java

```diff
@@ -6,30 +6,40 @@
 import pl.allegro.tech.hermes.common.message.wrapper.CompositeMessageContentWrapper;
 import pl.allegro.tech.hermes.common.message.wrapper.UnsupportedContentTypeException;
 import pl.allegro.tech.hermes.common.message.wrapper.UnwrappedMessageContent;
+import pl.allegro.tech.hermes.consumers.consumer.receiver.SchemaExistenceEnsurer;
 
 class BasicMessageContentReader implements MessageContentReader {
 
     private final CompositeMessageContentWrapper compositeMessageContentWrapper;
     private final KafkaHeaderExtractor kafkaHeaderExtractor;
     private final Topic topic;
+    private final SchemaExistenceEnsurer schemaExistenceEnsurer;
 
     BasicMessageContentReader(CompositeMessageContentWrapper compositeMessageContentWrapper,
                               KafkaHeaderExtractor kafkaHeaderExtractor,
-                              Topic topic) {
+                              Topic topic, SchemaExistenceEnsurer schemaExistenceEnsurer) {
         this.compositeMessageContentWrapper = compositeMessageContentWrapper;
         this.kafkaHeaderExtractor = kafkaHeaderExtractor;
         this.topic = topic;
+        this.schemaExistenceEnsurer = schemaExistenceEnsurer;
     }
 
     @Override
     public UnwrappedMessageContent read(ConsumerRecord<byte[], byte[]> message, ContentType contentType) {
         if (contentType == ContentType.AVRO) {
             Integer schemaVersion = kafkaHeaderExtractor.extractSchemaVersion(message.headers());
             Integer schemaId = kafkaHeaderExtractor.extractSchemaId(message.headers());
```

Comment:
For the sake of consistency, we should probably also wait for a schema when its ID is provided.